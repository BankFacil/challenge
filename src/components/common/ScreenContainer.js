import {SafeAreaView} from 'react-native';
import styled from 'styled-components';
import theme from '../theme';

const ScreenContainer = styled(SafeAreaView)`
  flex: 1;
  padding: ${p => p.padding ? theme.spacing.padding : 0}px;
  background: #ffffff;
`;

export default ScreenContainer;
