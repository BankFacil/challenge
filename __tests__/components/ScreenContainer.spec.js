import React from 'react';
import renderer from 'react-test-renderer';

import ScreenContainer from '../../src/components/common/ScreenContainer';

describe('ScreenContainer', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const tree = renderer.create(<ScreenContainer />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match to snapshot - with padding', () => {
      const tree = renderer.create(<ScreenContainer padding />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
