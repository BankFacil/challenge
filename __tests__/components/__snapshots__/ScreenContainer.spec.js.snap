// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenContainer Rendering should match to snapshot - with padding 1`] = `
<RCTSafeAreaView
  className="sc-bdVaJa lkMMd"
  emulateUnlessSupported={true}
  padding={true}
/>
`;

exports[`ScreenContainer Rendering should match to snapshot 1`] = `
<RCTSafeAreaView
  className="sc-bdVaJa kiIDIj"
  emulateUnlessSupported={true}
/>
`;
